src/dmap_import/util_api.py

```diff
@@ -0,0 +1,157 @@
+    )
+    download_log.log_start()
+
+    with requests.get(url, stream=True, timeout=60 * 5) as response:
+        # raise HTTP error, if occured
+        response.raise_for_status()
+        with open(local_path, "wb") as local_file:
+            for file_chunk in response.iter_content(chunk_size=None):
+                # chunk_size=None will write chunks in whatever size they are recieved
+                local_file.write(file_chunk)
+
+    file_size_mb = os.path.getsize(local_path) / (1024 * 1024)
+    download_log.add_metadata(file_size_mb=f"{file_size_mb:.4f}")
+    download_log.log_complete()
+
+    return local_path
+
+
+def get_api_results(url: str, db_manager: DatabaseManager) -> List[ApiResult]:
+    """
+    Execute GET request against CUBIC API URL using last_updated param to
+    filter results based on last_updated timestamp from ApiMetadata DB table
+
+    will throw if GET request result for 'success' is not True
+
+    :param url: full URL of CUBIC API Endpoint
+
+    :return list of filtered and sorted CUBIC API Results
+
+    All CUBIC API Endpoints appear to offer the following paramters:
+        - apikey: str:  API Authentication
+        - start_date: str: filter by date range <YYYY-MM-DD>
+        - end_date: str: filter by date range <YYYY-MM-DD>
+        - limit: int: reduce feteched result count
+        - offset: int: skip n records
+        - last_updated: str: filter by last updated <YYYY-MM-DD>
+
+    API Endpoints appear to produce a maximum of 100 records,
+    any `limit` value > 100 is ignored.
+
+    `offset` parameters appears to be non-functional, utilizing `offset` had no
+    impact on returned results during API testing
+
+    """
+    api_results_log = ProcessLogger(
+        "get_api_results",
+        url=url,
+    )
+    api_results_log.log_start()
+
+    last_update_query = sa.select(ApiMetadata.last_updated).where(
+        ApiMetadata.url == url,
+    )
+    db_result = db_manager.select_as_list(last_update_query)
+
+    # add params to GET request
+    # last_updated if last_update_dt available from ApiMetadata table
+    # apikey based on contents of url endpoint string
+    params = {}
+    if db_result:
+        last_updated_dt: datetime.datetime = db_result[0]["last_updated"]
+        params["last_updated"] = (
+            last_updated_dt.date() - datetime.timedelta(days=1)
+        ).strftime("%Y-%m-%d")
+        api_results_log.add_metadata(
+            last_updated_dt=last_updated_dt.isoformat()
+        )
+
+    if "datasetpublicusersapi" in url:
+        params["apikey"] = os.getenv("PUBLIC_KEY", "")
+    if "datasetcontrolleduserapi" in url:
+        params["apikey"] = os.getenv("CONTROLLED_KEY", "")
+
+    # execture GET request from CUBIC API Endpoint
+    # will throw if 200 status_code not recived
+    # if "success" attribute of response is not True will log and throw
+    response = requests.get(url, params=params, timeout=60)
+    response.raise_for_status()
```

Comment:
suggestion: you might want to check the status manually, as the empty response seems to return a 503 with `success: False`

Reply:
updated `status_code` handling and added retry logic